File: app/src/main/java/com/jrvermeer/psalter/Infrastructure/SimpleStorage.java
package com.jrvermeer.psalter.Infrastructure;

import android.content.Context;
import android.content.SharedPreferences;
import android.support.annotation.StringRes;

import com.jrvermeer.psalter.R;
import com.jrvermeer.psalter.UI.PsalterApplication;

public class SimpleStorage {
    public SimpleStorage(){
        c = PsalterApplication.getContext();
        sPref = c.getSharedPreferences("settings", Context.MODE_PRIVATE);

    }
    private SharedPreferences sPref;
    private Context c;

    public boolean getBoolean(@StringRes int id){
        return sPref.getBoolean(c.getString(id), false);
    }
    public void setBoolean(@StringRes int id, boolean val){
        sPref.edit().putBoolean(c.getString(id), val).apply();
    }
    public boolean toggleBoolean(@StringRes int id){
        boolean newVal = !getBoolean(id);
        setBoolean(id, newVal);
        return newVal;
    }

    private int getInt(@StringRes int id){
        return sPref.getInt(c.getString(id), 0);
    }
    private void setInt(@StringRes int id, int val){
        sPref.edit().putInt(c.getString(id), val).apply();
    }

    public boolean isNightMode(){
        return getBoolean(R.string.pref_nightmode);
    }
    public boolean toggleNightMode(){
        return toggleBoolean(R.string.pref_nightmode);
    }

    public boolean scoreShown(){
        return getBoolean(R.string.pref_showScore);
    }
    public boolean toggleScore(){
        return  toggleBoolean(R.string.pref_showScore);
    }

    public int getPageIndex(){
        return getInt(R.string.pref_lastindex);
    }
    public void setPageIndex(int i){
        setInt(R.string.pref_lastindex, i);
    }
}

File: app/src/main/java/com/jrvermeer/psalter/infrastructure/SimpleStorage.kt
package com.jrvermeer.psalter.infrastructure

import android.content.Context
import android.content.SharedPreferences
import android.support.annotation.StringRes

import com.jrvermeer.psalter.R

class SimpleStorage(private val context: Context) {
    private val sPref: SharedPreferences = context.getSharedPreferences("settings", Context.MODE_PRIVATE)

    val isNightMode get() = getBoolean(R.string.pref_nightmode)
    val isScoreShown get() = getBoolean(R.string.pref_showScore)

    var pageIndex
        get() = getInt(R.string.pref_lastindex)
        set(i) = setInt(R.string.pref_lastindex, i)

    fun getBoolean(@StringRes id: Int): Boolean {
        return sPref.getBoolean(context.getString(id), false)
    }

    fun setBoolean(@StringRes id: Int, `val`: Boolean) {
        sPref.edit().putBoolean(context.getString(id), `val`).apply()
    }

    fun toggleBoolean(@StringRes id: Int): Boolean {
        val newVal = !getBoolean(id)
        setBoolean(id, newVal)
        return newVal
    }

    private fun getInt(@StringRes id: Int): Int {
        return sPref.getInt(context.getString(id), 0)
    }

    private fun setInt(@StringRes id: Int, `val`: Int) {
        sPref.edit().putInt(context.getString(id), `val`).apply()
    }

    fun toggleNightMode() {
        val nightMode = toggleBoolean(R.string.pref_nightmode)
        Logger.changeTheme(nightMode)
    }

    fun toggleScore(): Boolean {
        val shown = toggleBoolean(R.string.pref_showScore)
        Logger.changeScore(shown)
        return shown
    }
}
